app/forms/catkey_form.rb

```diff
@@ -1,22 +1,89 @@
 # frozen_string_literal: true
 
-class CatkeyForm < ApplicationChangeSet
-  property :catkey, virtual: true
+class CatkeyForm < Reform::Form
+  collection :catkeys, prepopulator: ->(*) { catkeys << CatkeyForm::Row.new(value: '', refresh: true) if catkeys.size.zero? },
+                       populator: lambda { |collection:, index:, **|
+                                    if item = collection[index] # rubocop:disable Lint/AssignmentInCondition
+                                      item
+                                    else
+                                      collection.insert(index, CatkeyForm::Row.new)
+                                    end
+                                  } do
+    property :value
+    property :refresh
+    property :_destroy, virtual: true
+  end
+
+  validate :catkeys_acceptable
+
+  ### classes that define a virtual catkey model object and data structure, used in form editing...persistence is in the cocina model
+  class ModelProxy
+    def initialize(id:, catkeys:)
+      @id = id # the object ID
+      @catkeys = catkeys # the array of catkey objects (defined in custom class below)
+    end
+
+    attr_reader :id, :catkeys
+
+    def to_param
+      @id
+    end
+
+    def persisted?
+      true
+    end
+  end
+
+  class Row
+    attr_accessor :value, :refresh, :id
 
-  # When the object is initialized, copy the properties from the cocina model to the form:
-  def setup_properties!(_options)
-    self.catkey = Catkey.symphony_links(model).join(', ')
+    def initialize(attrs = {})
+      @id = attrs[:value]
+      @value = attrs[:value]
+      @refresh = attrs[:refresh]
+    end
+
+    def persisted?
+      id.present?
+    end
+
+    # from https://github.com/rails/rails/blob/f95c0b7e96eb36bc3efc0c5beffbb9e84ea664e4/activerecord/lib/active_record/nested_attributes.rb#L382-L384
+    def _destroy; end
+  end
+  ###
+
+  def catkeys_acceptable
+    # at most one catkey (not being deleted) can be set to refresh == true
+    errors.add(:refresh, 'is only allowed for a single catkey.') if catkeys.count { |catkey| catkey.refresh == 'true' && catkey._destroy != '1' } > 1
+    # must match the expected pattern
+    errors.add(:catkey, 'must be in an allowed format') if catkeys.count { |catkey| catkey.value.match(/^\d+(:\d+)*$/).nil? }.positive?
   end
 
-  # @raises [Dor::Services::Client::BadRequestError] when the server doesn't accept the request
-  # @raises [Cocina::Models::ValidationError] when given invalid Cocina values or structures
-  def save_model
-    return unless changed?(:catkey)
+  def save
+    cocina_object = Repository.find(@model.id)
```

Comment:
This causes a second load, which should be unnecessary, since we already loaded this in the controller.